web/controllers/session_controller.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Community.SessionController do
+  use Community.Web, :controller
+
+  def new(conn, _params) do
+    conn
+    |> render(:new)
+  end
+
+  def create(conn, %{"session" => %{"password" => password}}) do
+    if password == System.get_env("ADMIN_KEY") do
+      conn
+      |> put_session(:admin_key, password)
+      |> put_flash(:info, "Welcome home!")
+      |> redirect(to: redirect_path(conn))
+    else
+      :timer.sleep(1000)
+      conn
+      |> put_flash(:error, "NOPE")
+      |> redirect(to: "/")
+    end
+  end
+
+  defp redirect_path(conn) do
+    case get_session(conn, :redirect_path) do
+      nil -> "/"
```

Comment:
I would agree except that I don't assigning the variable our during the conditional.